pulsar-client-cpp/lib/CMakeLists.txt

```diff
@@ -113,5 +120,10 @@ else()
     install(TARGETS pulsarStatic DESTINATION lib)
 endif(LINK_STATIC)
 
-install(TARGETS pulsarShared pulsarStatic DESTINATION lib)
+if (BUILD_SHARED)
+    install(TARGETS pulsarShared pulsarStatic DESTINATION lib)
+else()
+    install(TARGETS pulsarStatic DESTINATION lib)
```

Comment:
the static lib should be built even if `BUILD_SHARED=TRUE`

Reply:
My goal was to enable a "self-contained" build, meaning zero dependencies on the currently installed system libraries, without going through whole Docker -> RPM process. My current build procedure builds all static libs for all of my external dependencies from their latest sources and links my executable to them. The shared library cannot be built in this environment since it depends on the other shared libs being present, or at least that the static libs be built with PIC which I'd rather not use. Enabling a "static-only" build resolves this issue for me.

Reply:
Gotcha, so you basically need to use LINK_STATIC and disable the shared library build. 

I think it might work even by just doing `make pulsarStatic` instead of the general `make`. 

In any case, for the if/else, this would be changing the default build by only building the shared library by default and not the static library. I think intuitively, `BUILD_SHARED=TRUE` shouldn't necessary mean that static library is not built. If we want to disable that, `BUILD_STATIC=..` could be added.

I mean: 

```
if (BUILD_SHARED)
    install(TARGETS pulsarShared pulsarStatic DESTINATION lib)
endif()

install(TARGETS pulsarStatic DESTINATION lib)
```

Reply:
Oh.. I actually mis-read the code above.. Could it be changed to 

```
if (BUILD_SHARED)
    install(TARGETS pulsarShared DESTINATION lib)
endif()

install(TARGETS pulsarStatic DESTINATION lib)
```

That will make the if/else more evident

Reply:
So it looks like calling the pulsarStatic target directly will accomplish what I need. Thanks, sorry I missed that.